cdap-common/src/main/java/co/cask/cdap/common/internal/guava/ClassPath.java

```diff
@@ -136,7 +136,10 @@ public static ClassPath from(URI uri, ClassLoader classLoader) throws IOExceptio
    * @since 16.0
    */
   public ImmutableSet<ClassInfo> getAllClasses() {
-    return FluentIterable.from(resources).filter(ClassInfo.class).toImmutableSet();
+    return resources.stream()
+      .filter(ClassInfo.class::isInstance)
+      .map(resourceInfo -> (ClassInfo) resourceInfo)
+      .collect(Collectors.collectingAndThen(Collectors.toSet(), ImmutableSet::copyOf));
```

Comment:
Use `Collectors.toCollection(LinkedHashSet::new)` instead fo preserve the behavior, which an `ImmutableSet` has iteration order to same as insertion order.